AudioKitSynthOne/DSP/AudioRecorder.swift

```diff
@@ -62,13 +64,12 @@ class AudioRecorder {
                 callback: { exportedFile, error in
                     if error != nil { return }
                     guard let file = exportedFile else { return }
-                    DispatchQueue.main.async {
-                        self.fileDelegate?.didFinishRecording(file: file)
+                    DispatchQueue.main.async { [unowned self] in
+                        guard let fileDelegate = self.fileDelegate else { return }
+                        fileDelegate.didFinishRecording(file: file)
                         self.updateView()
                     }
             })
-            viewTimer?.invalidate()
-            viewDelegate?.updateRecorderView(state: .Exporting, time: 0)
```

Comment:
Can you explain why you made these changes? 
Technically these are wrong I think, because now you set the UI to Exporting, even though that could fail (in the subsequent guard statements...